### ISSUE TITLE
LO 7.4 Writer specific doc structure+specific user dynamic action with LO induces LT fail conditions

### ISSUE THREAD

**Issue by BloomingAzaleas** [languagetool-org/languagetool#7089]
A specific, static Writer document structure using the Writer 'Section' feature combined with a specific dynamic sequence of LO operations not nominally related to LT induces an LT Recheck Document failure condition. The failure condition is either a premature LT S&G scan completion condition shown in the S&G pane, or an LT error message box followed by the premature scan completion display.

A .zip upload containing a detailed writeup document and a demo file set for the issue to follow.

**Comment by BloomingAzaleas** [comment]
.zip file containing a README write up and a set of directories containing test cases that, oddly, mostly demonstrate LT expected behavior in the near neighborhood of the specific failure conditions, then two cases of LT fail.

[LT-7089-README+Demo-set.zip](https://github.com/languagetool-org/languagetool/files/9548621/LT-7089-README%2BDemo-set.zip)

**Comment by FredKruse** [comment]
Thank you for your systematic tests.
The last commit solves the bugs documented in your cases 130 and 140.
Please test it with the tomorrow nightly.

**Comment by BloomingAzaleas** [comment]
I should be able to test over the weekend.

Regards.

**Comment by BloomingAzaleas** [comment]
Tested with LanguageTool-20220916-snapshot.oxt and
Version: 7.4.0.3 (x64) / LibreOffice Community
Build ID: f85e47c08ddd19c015c0114a68350214f7066f5a
CPU threads: 8; OS: Windows 10.0 Build 19044; UI render: Skia/Vulkan; VCL: win
Locale: en-US (en_US); UI: en-US
Calc: CL

Works as expected. Bravo.

At first I was chasing textboxes and frames for this thing until bisecting eliminated those, at which point I noticed, what the heck? Footnotes? Circled that drain for a while until I noticed the only other object was a Section. The set of conditions to trigger seemed so random. Had to let it rest now and then because so tedious to formulate and execute elimination tests.

Side note: This LT snap both scans and will successfully modify appendnotes originating in an RO Section even though a manual modification attempt via the LO GUI will fail. Given that a Section COULD BE marked RW (effectively a pass-through of the edit action to the source document), this LT behavior is actually useful if documented that RO Section edits via LT are not persistent. Documented, documented,... hmmmm have I ranted about LT documentation before?

Regards.
